components/brave_rewards/browser/rewards_service_factory.cc

```diff
@@ -22,6 +29,20 @@
 #include "brave/components/brave_rewards/browser/rewards_service_impl.h"
 #endif
 
+namespace {
+#if BUILDFLAG(BRAVE_REWARDS_ENABLED)
+void OverridePrefsForPrivateProfileUserPrefs(Profile* profile) {
+  if (profile->GetProfileType() == Profile::REGULAR_PROFILE)
+    return;
+
+  // rewards button should be hidden on guest and tor profile.
+  PrefService* pref_service = profile->GetPrefs();
+  pref_service->SetBoolean(kHideBraveRewardsButton, true);
+  pref_service->SetBoolean(brave_rewards::prefs::kBraveRewardsEnabled, false);
```

Comment:
Would a user still be able to visit brave://rewards from incognito and re-enable it? Not sure if that's a problem, I guess there's only so much we can do.

Reply:
AFAIK, rewards service doesn't work with incognito profile 
 and `RewardsServiceFactory::GetForProfile` just return `nullptr`.
When user loading `brave://rewards` page in incognito, it is redirected to normal profile's window.
So, I think we can completely disable rewards service in non-normal profile.

Reply:
OK, makes sense.